### PR TITLE
Reimplement license inspect command and refactor assign + config loading

### LINKED ISSUE #1131: License inspect command
License inspect command should fetch license jwt token from kube-api and display license details as below:

In case we know company name (`data.owner.company`):

```
<data.name> for up to <data.maxnodes> nodes
Licensed to <data.owner.company>.
Valid until <data.valid_until>  // formatted as December 24, 2019
```
In case we don't know company name but we know the full name

```
Kontena Pharos PRO License for up to 10 nodes
Licensed to <data.owner.first_name> <data.owner.last_name>
Valid until December 24, 2019
```
In case we don't know company name or full name

```
Kontena Pharos PRO License for up to 10 nodes
Licensed to <owner.username>
Valid until December 24, 2019
```

In case of evaluation (license not found):

```
Kontena Pharos PRO Evaluation License
Valid until March 2, 2019 (17 days remaining)
Please contact eval@kontena.io to extend your evaluation
```

### PR DESCRIPTION
**Pull request by kke**
Fixes #1131 

Re-refactors some of the refactorings made in #1147

### CODE REVIEW

File: non-oss/pharos_pro/commands/license_inspect_command.rb [hunk 1]
```diff
@@ -2,5 +2,117 @@
...
+      Base64.decode64(encoded_jwt)
+    end
+
+    def default_cluster_id
+      prepare_config
+      cluster_context['cluster-id'] || signal_error('Failed to get cluster id')
+    end
+
+    def execute
+      puts [license_type, owner, valid_until, contact_note].compact
```
Comment (kke) [inline non-oss/pharos_pro/commands/license_inspect_command.rb]: A lot of stuff had to be added to implement the requested display format in the issue.

File: non-oss/pharos_pro/license_key.rb [hunk 2]
```diff
@@ -49,8 +70,7 @@ def validate
     def decode_payload
       JSON.parse(Base64.decode64(payload)).fetch('data')
     rescue StandardError => ex
-      errors << "Can't decode token payload (#{ex.class.name} : #{ex.message})"
-      {}
+      { 'errors' => ["Can't decode token payload (#{ex.class.name} : #{ex.message})"] }
```
Comment (kke) [inline non-oss/pharos_pro/license_key.rb]: This fixes an infinite loop / stack overflow when payload is invalid.

File: non-oss/pharos_pro/commands/license_inspect_command.rb [hunk 3]
```diff
@@ -2,5 +2,117 @@
...
+    def owner
+      return if license_key.nil?
+
+      "Licensed to #{license_key.owner}"
+    end
+
+    def contact_note
+      return if license_key
+
+      "Please contact eval@kontena.io to extend your evaluation"
```
Comment (nevalla) [inline non-oss/pharos_pro/commands/license_inspect_command.rb]: Please change `eval@kontena.io` to `sales@kontena.io`

Comment (kke) [reply non-oss/pharos_pro/commands/license_inspect_command.rb]: Changed. This info is only displayed when there's no license, but there's no contact info / instructions when license has expired or is near expiration or something.

**Comment by kke** [comment]
Would be nice to have a way to silence the cluster manager / phases / sharpening tools stuff.

**Pull request merged**